pkg/phlaredb/symdb/resolver.go

```diff
@@ -52,12 +67,15 @@ type lazyPartition struct {
 	done    chan struct{}
 }
 
-func NewResolver(ctx context.Context, s SymbolsReader) *Resolver {
+func NewResolver(ctx context.Context, s SymbolsReader, opts ...ResolverOption) *Resolver {
```

Comment:
that is a nice way to make this more extensible, good stuff!

Reply:
Thanks to Rob Pike https://commandcenter.blogspot.com/2014/01/self-referential-functions-and-design.html :)